### PR TITLE
[MM-40467] Fix panic if oauth2user is empty

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
If the oauth2 user is not set, the app should not try to access the data. I'm proposing a defensive approach here, which every time checks if the data is there.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40467

### CODE REVIEW

**Review by jfrerich — APPROVED** [review]

File: src/utils/utils.ts [hunk 1]
```diff
@@ -164,7 +164,7 @@ export function isUserSystemAdmin(actingUser: UserProfile): boolean {
 }
 
 export function isConnected(oauth2user: StoredOauthUserToken): boolean {
-    if (oauth2user && oauth2user.token.access_token && oauth2user.token.access_token !== '') {
+    if (oauth2user?.token?.access_token !== '') {
```
Comment (mickmister) [inline src/utils/utils.ts:167]: In this case `access_token` will be `undefined`, and we will incorrectly assumed it's defined when we assert that it's not the empty string. An empty string is [treated as falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) so the check below will work:

```suggestion
    if (oauth2user?.token?.access_token) {
```

@jfrerich Any reason for the explicit empty string comparison in the existing code?

**Review by mickmister — CHANGES_REQUESTED** [review]
Just one request to do a `falsy` comparison rather than an explicit empty string comparison on one of the checks.

Also, should the code following the spots modified in this PR be changed to react to the fact that the value may be undefined?

**Comment by mickmister** [comment]
Currently blocking on my comment:

> Also, should the code following the spots modified in this PR be changed to react to the fact that the value may be undefined?

The current change in the PR may result in panics further down in the functions that were edited, or functions that receive these objects containing the edited values. @jfrerich Can you please take a look at the code paths here?

**Comment by hanzei** [comment]
> The current change in the PR may result in panics further down in the functions that were edited, or functions that receive these objects containing the edited values. @jfrerich Can you please take a look at the code paths here?

@mickmister I don' think there is an issue here. The access token is only being used in `newZDClient`, which checks if it's set: https://github.com/mattermost/mattermost-app-zendesk/blob/042d0e50d0f60949e28eda0e9030ca4632676f65/src/clients/zendesk.ts#L67-L69

Calls to `newZDClient` are mostly guarded with a `catch`. The only one missing a catch is `fCreateTarget`. Should we add that?

**Comment by mickmister** [comment]
@hanzei Thanks for going through the code more thoroughly to check for other issues :+1:

> The only one missing a catch is `fCreateTarget`. Should we add that?

If the improvement is easy to add, sure why not add it here. The original problem here is only with the OAuth complete call (I think) so it's not necessary to introduce the error catch in this PR, but it doesn't hurt to add more defensive code when we see the opportunity.

**Comment by hanzei** [comment]
@mickmister Added a catch around it. Maybe it's worth opening a (HW?) ticket to add a `catch` for all `CallResponseHandler`'s?

**Comment by hanzei** [comment]
@mickmister Gentle reminder to review this PR

**Review by mickmister — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Heads up that I'm going to merge this PR as it is as it's it only adds more defensive code and error checks and blocks the review on https://github.com/mattermost/mattermost-app-zendesk/pull/90.

**Pull request merged**